src/core/ThemeProvider/ThemeProvider.test.tsx

```diff
@@ -25,19 +25,69 @@ afterEach(() => {
 });
 
 it('should respect os theme (light)', () => {
-  window.matchMedia = jest.fn().mockReturnValue({ matches: false });
+  window.matchMedia = jest.fn().mockReturnValueOnce({ matches: false });
 
   render(<ThemeProvider theme='os' />);
   expectLightTheme();
 });
 
 it('should respect os theme (dark)', () => {
-  window.matchMedia = jest.fn().mockReturnValue({ matches: true });
+  window.matchMedia = jest.fn().mockReturnValueOnce({ matches: true });
 
   render(<ThemeProvider theme='os' />);
   expectDarkTheme();
 });
 
+it('should observe changes to os theme', () => {
+  const originalMatchMedia = window.matchMedia;
+  const listeners: Array<(e: { matches: boolean }) => void> = [];
+  let matches = false;
+
+  const mockDispatch = jest
+    .fn()
+    .mockImplementation((event: { matches: boolean }) => {
+      listeners.forEach((listener) => {
+        matches = event.matches;
+        listener(event);
+      });
+      return true;
+    });
+
+  window.matchMedia = jest.fn().mockReturnValueOnce({
+    matches: matches,
+    addEventListener: (
+      _: 'change',
+      listener: (e: { matches: boolean }) => void,
+    ) => {
+      listeners.push(listener);
+    },
+    removeEventListener: (
```

Comment:
Maybe also check that `removeEventListener` was called.

Reply:
Added